tests/osgi-tests/src/test/java/org/wso2/carbon/metrics/osgi/MetricsTest.java

```diff
@@ -78,77 +87,90 @@ private Bundle getBundle(String name) {
         return bundle;
     }
 
+    @Configuration
+    public Option[] createConfiguration() {
+        return new Option[]{
+                copyOSGiLibBundle(maven()
+                        .artifactId("org.wso2.carbon.metrics.sample.service")
+                        .groupId("org.wso2.carbon.metrics")
+                        .versionAsInProject()),
+                copyOSGiLibBundle(maven()
+                        .artifactId("org.wso2.carbon.metrics.sample.consumer")
+                        .groupId("org.wso2.carbon.metrics")
+                        .versionAsInProject())
+        };
+    }
     @Test
     public void testMetricsCoreBundle() {
         Bundle coreBundle = getBundle("org.wso2.carbon.metrics.core");
         Assert.assertEquals(coreBundle.getState(), Bundle.ACTIVE);
     }
 
-    @Test
+    @Test(dependsOnMethods = { "testMetricsCoreBundle" })
```

Comment:
What's the reason for adding "dependsOnMethods"? We usually should avoid depending on other methods as much as possible.